Reduce the number of states

guillaumerose opened the pull request:
None is almost always an alias of Error. Also, it is a real mystery to understand why a status is None. Let's remove it. 

Also treat unexpected libvirt statuses as errors like pause, pmsuspended, etc.
crc doesn't nothing with state.Saved, Paused, etc. It is better to remove it and show an error.

guillaumerose committed:
Remove state.None, use state.Error instead.

None is all the time an alias of Error. This change makes it clear in
the code.

guillaumerose committed:
Domain is not expected to be paused or pmsuspended.

If this happens, this is treated as an error. The user did probably
something with the VM manually.

cfergeau reviewed (commented):
There is at least one situation when the VM can be paused without user intervention, it's when the filesystem where ~/.crc/machines/crc/crc.qcow2 is located is full. This can be detected by looking at the error details (`DOMAIN_PAUSED_IOERROR`) but we are not there yet, so probably good enough to return an error.
`PAUSED` can also happen when the guest crashes (`DOMAIN_PAUSED_CRASHED`), which we've seen a few times in the past.
And I see a `DOMAIN_PAUSED_STARTING_UP` which makes me wonder if we could see a very transient `DOMAIN_PAUSED` state during startup if we poll the VM?

With all that said, looks good to me.

guillaumerose commented:
Yes you are right for paused. I ran a loop asking for the state of crc domain and I got:

```
$ repeat 1000 virsh --connect qemu:///system domstate crc
error: failed to get domain 'crc'

error: failed to get domain 'crc'

paused

paused

paused

paused

paused

paused

paused

paused

running

running

running
```

guillaumerose committed:
When starting the VM, the state is PAUSED for a very short time

When doing a crc start, `virsh --connect qemu:///system domstate crc --reason`
shows these transitions:

1. error: failed to get domain 'crc'
2. paused (starting up)
3. running (booted)

guillaumerose committed:
SHUTDOWN means the VM is in the process of being shutdown

See https://github.com/code-ready/machine-driver-libvirt/issues/40
